File: app/src/main/java/amrabed/android/release/evaluation/data/tables/TaskTable.java
package amrabed.android.release.evaluation.data.tables;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import amrabed.android.release.evaluation.data.entities.Task;

@Dao
public interface TaskTable {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertTasks(Task... tasks);

    @Update
    void updateTask(Task task);

    @Delete
    void deleteTask(Task task);

    @Query("SELECT * FROM tasks")
    LiveData<List<Task>> loadCurrentTasks();
}

File: app/src/main/java/amrabed/android/release/evaluation/data/tables/TaskTable.kt
package amrabed.android.release.evaluation.data.tables

import amrabed.android.release.evaluation.core.Task
import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface TaskTable {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertTasks(vararg tasks: Task?)

    @Update
    fun updateTask(task: Task?)

    @Delete
    fun deleteTask(task: Task?)

    @Query("SELECT * FROM tasks ORDER BY currentIndex ASC")
    fun loadCurrentTasks(): LiveData<MutableList<Task>>?

    @Query("SELECT * FROM tasks WHERE reminder IS NOT NULL ORDER BY currentIndex ASC")
    fun loadReminders(): LiveData<MutableList<Task>>?
}